### ISSUE TITLE
SetPropertiesDefaults errors if masterProfile.count > 1

### ISSUE THREAD

**Issue by mboersma** [Azure/aks-engine#991]
**Is this an ISSUE or FEATURE REQUEST?** (choose one):

ISSUE

---

**What version of aks-engine?**:

Current master, post v0.33.5.

---

I noticed after #947 merged that we weren't checking the error return from `SetPropertiesDefaults` in some tests. After changing the code to handle it, the test kubernetes.json apimodel for azurestack throws this error unless `masterProfile.count` is set to 1:
```
The raw pem is not a valid PEM formatted block:
caCertificate
```

cc: @honcao

**Comment by CecileRobertMichon** [comment]
could this be related to https://github.com/Azure/aks-engine/issues/883 by any chance?

**Comment by mboersma** [comment]
This was not actually fixed by #1053: The test changes in #992 will still fail if masterCount > 1, so that needs to be root-caused and fixed separately, rather than using the workaround in that PR.
